test_pull_requests

```diff
@@ -546,13 +583,44 @@ module Hub
       return pull_request, bot_comment, comments
     end
 
-    def local_merge_pull_request(pull_id, repo)
-      $stderr.puts "Local merging pull request ##{pull_id} for repo '#{repo}'"
+    def can_skip_merge_tests(pull_id, repo, settings)
+      preceding_test_settings = Properties['settings'][settings['preceding_test_settings']]
+      if preceding_test_settings.nil?
+        return false
+      end
+      $stderr.puts "  Checking if tests are required to merge pull request ##{pull_id} for repo '#{repo}'"
+      p_test_prefix = preceding_test_settings['test_prefix']
+      comments = get_comments(pull_id, repo)
+      bot_comment = get_comment_with_prefix(pull_id, repo, p_test_prefix, comments)
+      raise "Missing '#{p_test_prefix}' comment!" if bot_comment.nil?
+      pull_request = get_pull_request(pull_id, repo, 10)
+      raise "Pull request isn't open!" unless pull_request['state'] == 'open'
+      comment = bot_comment['body']
+      build_url, extended_tests, comment_base_commit = extract_bot_comment_metadata(comment, p_test_prefix, settings)
+      if comment !~ /^#{p_test_prefix} SUCCESS/
+        return false
+      end
+      comment_base_commit = base_commit_tag_to_sha(comment_base_commit)
+      base_commit = pull_request['base']['sha']
+      return base_commit == comment_base_commit, build_url
```

Comment:
@stevekuznetsov good point, that is kind of extremely hackish. I'll try to find a way to make that better...